src/Materials/Node/Blocks/Dual/textureBlock.ts

```diff
@@ -284,22 +286,24 @@ export class TextureBlock extends NodeMaterialBlock {
         }
 
         if (!this.texture || !this.texture.getTextureMatrix) {
-            defines.setValue(this._defineName, false);
-            defines.setValue(this._mainUVDefineName, true);
+            if (this._isMixed) {

```

Comment:
Can you elaborate on why we need this test?

Reply:
Because else we can get a `#define undefined` entry in the list of defines because when `_isMixed == false` we don't execute `_injectVertexCode` which is where `_defineName` and `_mainUVDefineName` are set.